hosted/gke/p1/p1_import_test.go

```diff
@@ -87,6 +88,76 @@ var _ = Describe("P1Importing", func() {
 			_, err = helper.AddNodePool(cluster, ctx.RancherAdminClient, 1, "WINDOWS_LTSC_CONTAINERD", true, true)
 			Expect(err).To(BeNil())
 		})
+
+		It("should not be able to update the only non-windows nodepool to windows", func() {
+			testCaseID = 264
+
+			upgradedCluster := new(management.Cluster)
+			upgradedCluster.Name = cluster.Name
+			upgradedCluster.GKEConfig = cluster.GKEConfig
+
+			updateNodePoolsList := cluster.GKEConfig.NodePools
+			for i := 0; i < len(updateNodePoolsList); i++ {
+				updateNodePoolsList[i].Config.ImageType = "WINDOWS_LTSC_CONTAINERD"
+			}
+
+			upgradedCluster.GKEConfig.NodePools = updateNodePoolsList
+
+			_, err := ctx.RancherAdminClient.Management.Cluster.Update(cluster, &upgradedCluster)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("at least 1 Linux node pool is required"))
+		})
+	})
+
+	When("a cluster is created with at least 2 node pools", func() {
+		var cluster *management.Cluster
+
+		BeforeEach(func() {
+			var err error
+			err = helper.CreateGKEClusterOnGCloud(zone, clusterName, project, k8sVersion)
+			Expect(err).To(BeNil())
+
+			err = helper.AddNodePoolOnGCloud(clusterName, zone, project, "")
+			Expect(err).To(BeNil())
+
+			cluster, err = helper.ImportGKEHostedCluster(ctx.RancherAdminClient, clusterName, ctx.CloudCred.ID, zone, project)
+			Expect(err).To(BeNil())
+			cluster, err = helpers.WaitUntilClusterIsReady(cluster, ctx.RancherAdminClient)
+			Expect(err).To(BeNil())
+			// Workaround to add new Nodegroup till https://github.com/rancher/aks-operator/issues/251 is fixed
+			cluster.GKEConfig = cluster.GKEStatus.UpstreamSpec
+		})
+
+		AfterEach(func() {
+			if ctx.ClusterCleanup && cluster != nil {
+				err := helper.DeleteGKEHostCluster(cluster, ctx.RancherAdminClient)
+				Expect(err).To(BeNil())
+				err = helper.DeleteGKEClusterOnGCloud(zone, project, clusterName)
+				Expect(err).To(BeNil())
+			} else {
+				fmt.Println("Skipping downstream cluster deletion: ", clusterName)
+			}
+		})
+
+		It("For a given NodePool with a non-windows imageType, updating it to a windows imageType should fail", func() {
+			testCaseID = 55
+			upgradedCluster := new(management.Cluster)
+			upgradedCluster.Name = cluster.Name
+			upgradedCluster.GKEConfig = cluster.GKEConfig
+
+			updateNodePoolsList := cluster.GKEConfig.NodePools
+			updateNodePoolsList[0].Config.ImageType = "WINDOWS_LTSC_CONTAINERD"
+
+			upgradedCluster.GKEConfig.NodePools = updateNodePoolsList
+
+			cluster, err := ctx.RancherAdminClient.Management.Cluster.Update(cluster, &upgradedCluster)
+			Expect(err).To(BeNil())
```

Comment:
Could be a new function _UpdateNodePoolImage_

Reply:
I have added a generic UpdateFunction instead.